apps/heft/src/schemas/typescript.schema.json

```diff
@@ -90,6 +90,18 @@
             "type": "string",
             "pattern": "[^\\\\]"
           }
+        },
+
+        "destinationFolderName": {
```

Comment:
Should there be support for specifying an array of destination folders? This was previously supported via the `additionalModuleKindsToEmit.outFolderName` (which could have been multiple destinations).

That's something I can see being a feature request asked for in the future.

Reply:
My main goal was to be able to change the defaults (`src` and `lib`) to be customizable. The multiple outputs can be still be created with the "additional module kinds" I think. If it's a design decision to add the multiple outputs, I can add it.

Reply:
You're right on it still being supported by additional module kinds (I see [that here](https://github.com/microsoft/rushstack/pull/2315/files#diff-e8969b10cd8bf77fc91c043f56cda3c58d391b286b0bdde39608447163332b66R113)).  It's a design decision suggestion as I can see users wanting to copy over into multiple destinations.

Reply:
Not sure who may be able to guide us on the preferred design here. @iclanton, I am tagging you because you renamed the PR, not sure if you can give us guidance or point to somebody that can.

Reply:
@D4N14L is putting together a feature that touches a similar area here, so it would be good to try to align the nomenclature and design. I believe he is supporting multiple destination folders, but I understand why you made the decision you made here.

Maybe we can take this PR one step further and try to infer the source and out folders from the tsconfig, and if they can't be inferred, log something in `--verbose` mode and take overridden values from `typescript.json`. @manrueda, thoughts?

Reply:
I can work on inferring it from the `tsconfig.json` data and logs on error. In my case I need one to one copy, and it's from the source code folder to the output folder, so it works for me.

Reply:
@manrueda FWIW, here's my implementation of a generalized copy plugin, which you can use for schema guidance, though if you have any suggestions feel free to let me know! #2333

Reply:
I am back working on this, but I have some question about the best approach. Currently `CopyStaticAssetsPlugin.ts` does not have knowledge about `tsconfig.json`, the closest place where that is accessible is `TypeScriptBuilder.ts` but it's private and not exposed in any way. I can extract the logic from `TypeScriptBuilder.ts` and put it in the `utilities` section as a `TypeScript.ts` file for TS utilities, but not sure if that is desired. What are your suggestions?

Reply:
Given the intended purpose of `CopyStaticAssets`, shouldn't the task get its paths directly from the compiler configuration?  Whereas if the person wants to specify arbitrary source/destination folders, the new [copyFiles task](https://github.com/microsoft/rushstack/pull/2333) is probably a better way.

Reply:
That was my first approach, what is right now in this PR. The source and destination are defaulted to `src` and `lib` but can be customized in the `typescript.json` file. If that sounds good, and any more complex flow should be resolved by the `copyFiles` task, I can update the code to follow the naming as `copyFiles` and update the PR.